speed-dreams/source-2.2.3/src/libs/robottools/rthumandriver.cpp

```diff
@@ -1080,7 +1080,14 @@ static void common_drive(const int index, tCarElt* car, tSituation *s)
 
     // SIMULATED DRIVING ASSISTANCE: added if condition: only handle steering cmd input if the participant is allowed to.
     // Do not let the human control the car when the AI is in full control
-    if (SMediator::GetInstance()->GetPControlSettings().ControlSteer && SMediator::GetInstance()->GetInterventionType() != INTERVENTION_TYPE_COMPLETE_TAKEOVER)
+    bool canControlSteer = SMediator::GetInstance()->GetPControlSettings().ControlSteer &&
+                           SMediator::GetInstance()->GetInterventionType() != INTERVENTION_TYPE_AUTONOMOUS_AI;
+    if(SMediator::GetInstance()->GetInterventionType() == INTERVENTION_TYPE_COMPLETE_TAKEOVER && SMediator::GetInstance()->GetAllowedActions().Steer)
+    {
+        canControlSteer &= SMediator::GetInstance()->IsSteerDecision();
+    }
```

Comment:
This is getting a lot, maybe this can become a method in the mediator? SMediator::CanUserSteer()?
Same for brake and accel.